test/unit/resources/aws_ebs_volume_test.rb

```diff
@@ -0,0 +1,75 @@
+require 'helper'
+
+class TestEbs < Minitest::Test
+  Id = 'volume-id'.freeze
+
+  def setup
+    @mock_conn = Minitest::Mock.new
+    @mock_client = Minitest::Mock.new
+    @mock_resource = Minitest::Mock.new
+    @mock_iam_resource = Minitest::Mock.new
+
+    @mock_conn.expect :ec2_client, @mock_client
+    @mock_conn.expect :ec2_resource, @mock_resource
+    @mock_conn.expect :iam_resource, @mock_iam_resource
+  end
+
+  def test_that_id_returns_id_directly_when_constructed_with_an_id
+    assert_equal Id, AwsEbsVolume.new(Id, @mock_conn).id
+  end
+
+  def test_that_id_returns_fetched_id_when_constructed_with_a_name
+    mock_volume = Minitest::Mock.new
+    mock_volume.expect :nil?, false
+    mock_volume.expect :id, Id
+    @mock_resource.expect :volumes, [mock_volume], [Hash]
+    assert_equal Id, AwsEbsVolume.new({ name: 'cut' }, @mock_conn).id
```

Comment:
Ruby can accept method arguments like this, but we prefer `method(foo)` over `method foo`. Again, this isn't your fault. We are looking to add it to the style guide.

Example:

```
assert_equal(Id, AwsEbsVolume.new({ name: 'cut' }, @mock_conn).id)
```